scripts/apps/authoring-bridge/widgets/ai-widget/ai-assistant.tsx

```diff
@@ -0,0 +1,239 @@
+import HeadlinesTab from './headlines';
+import SummaryTab from './summary';
+import {extensions} from 'appConfig';
+
+// Can't call `gettext` in the top level
+const getLabel = () => gettext('AI Assistant');
+
+type IProps = React.ComponentProps<
+    IExtensionActivationResult['contributions']['authoringSideWidgets'][0]['component']
+>;
+
+interface IState {
+    headlinesOpen: boolean;
+    summaryOpen: boolean;
+
+    /**
+     * Handle loading of each request separately,
+     */
+    loadingHeadlines: boolean;
+    loadingSummary: boolean;
+
+    headlines: Array<string>;
+    error: boolean;
+    summary: string;
+}
+
+class AiAssistantWidget extends React.PureComponent<IProps, IState> {
+    aiAssistant: IExtensionActivationResult['contributions']['aiAssistant'];
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.aiAssistant = extensions['ai-assistant'].activationResult?.contributions?.aiAssistant;
+
+        this.state = {
+            headlinesOpen: false,
+            summaryOpen: false,
+            headlines: [],
+            error: false,
+            loadingSummary: true,
+            loadingHeadlines: true,
+            summary: '',
+        };
+
+        this.setError = this.setError.bind(this);
+        this.generateHeadlines = this.generateHeadlines.bind(this);
+        this.generateSummary = this.generateSummary.bind(this);
+    }
+
+    setError() {
+        this.setState({
+            error: true,
+        });
+    }
+
+    generateHeadlines() {
+        if (this.aiAssistant?.generateHeadlines == null) {
+            this.setError();
+        } else {
+            this.aiAssistant.generateHeadlines(this.props.article)
+                .then((res) => {
+                    this.setState({
+                        loadingHeadlines: false,
+                        headlines: res,
+                    });
+                }).catch(() => {
+                    this.setError();
+                });
+        }
+    }
+
+    generateSummary() {
+        if (this.aiAssistant?.generateSummary == null) {
+            this.setError();
+        } else {
+            this.aiAssistant.generateSummary(this.props.article)
+                .then((res) => {
+                    this.setState({
+                        loadingSummary: false,
+                        summary: res,
+                    });
+                }).catch(() => {
+                    this.setError();
+                });
+        }
+    }
+
+    render() {
+        return (
+            <AuthoringWidgetLayout
+                header={(
+                    <Spacer v gap="0" alignItems="center">
+                        <AuthoringWidgetHeading
+                            widgetName={getLabel()}
+                            editMode={false}
+                        />
+                        {(this.state.summaryOpen || this.state.headlinesOpen) && (
+                            <>
+                                <div className="p-1">
```

Comment:
It's a helper class from UIF. I'm not sure how you should be aware of that, I don't want to leave a comment on every class I use. Since we're aiming for reusability and control we have to use helper classes and no inline styles as much as possible.

Reply:
I was wondering if you left if from testing since name isn't much different from `a-1 b-2 c-3` :smile: